client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java

```diff
@@ -77,26 +88,29 @@
      *                          This is used if the controller endpoints are static and can be directly accessed.
      *                      2. pravega://ip1:port1,ip2:port2,...
      *                          This is used to autodiscovery the controller endpoints from an initial controller list.
+     * @param connectionFactory The factory for connections to the pravega Server
      */
-    public ControllerImpl(final URI controllerURI) {
+    public ControllerImpl(final URI controllerURI, ConnectionFactory connectionFactory) {
```

Comment:
If you like I could split it into two methods, one the computes all the successor segments to the StreamCut provided which would go on the controller. This could be easily optimized to a single call even without support for #1436. And leave the computation of the lengths (which requires the connectionFactory) in ReaderGroup.